app/src/main/java/org/mifos/mobile/ui/recent_transactions/RecentTransactionScreen.kt

```diff
@@ -0,0 +1,237 @@
+
+    LaunchedEffect(key1 = Unit) {
+        viewModel.loadRecentTransactions(false, 0)
+    }
+
+    RecentTransactionScreenContent(
+        uiState = uiState,
+        navigateBack = navigateBack,
+        retryConnection = { viewModel.loadRecentTransactions(false, 0) },
+        isRefreshing = isRefreshing,
+        onRefresh = { viewModel.refresh() }
+    )
+}
+
+@OptIn( ExperimentalMaterial3Api::class)
+@Composable
+fun RecentTransactionScreenContent(
+    uiState: RecentTransactionUiState,
+    navigateBack: () -> Unit,
+    retryConnection: () -> Unit,
+    isRefreshing: Boolean,
+    onRefresh: () -> Unit) {
+
+    val context = LocalContext.current
+    val pullRefreshState = rememberPullToRefreshState()
+
+        Box(Modifier.nestedScroll(pullRefreshState.nestedScrollConnection))
+        {
+            Column(modifier = Modifier.fillMaxSize()) {
+
+                when (uiState) {
+
+                    RecentTransactionUiState.EmptyTransaction -> {
+                        MifosErrorComponent(isEmptyData = true)
+                    }
+
+                    is RecentTransactionUiState.Error -> {
+                        MifosErrorComponent(
+                            isNetworkConnected = Network.isConnected(context),
+                            isEmptyData = false,
+                            isRetryEnabled = true,
+                            onRetry = retryConnection
+                        )
+                    }
+
+                    RecentTransactionUiState.Initial -> {
+                    }
+
+                    RecentTransactionUiState.Loading -> {
+
+                        MifosProgressIndicator(
+                            modifier = Modifier
+                                .fillMaxSize()
+                                .padding(10.dp)
+                                .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                        )
+                    }
+
+                    is RecentTransactionUiState.LoadMoreRecentTransactions -> {
+
+                    }
+
+                    is RecentTransactionUiState.RecentTransactions -> {
+
+                        if (uiState.transactions?.isNotEmpty() == true) {
+                            LoadRecentTransactions(transactionList = uiState.transactions as ArrayList<Transaction>)
+                        } else {
+                            MifosErrorComponent(isEmptyData = true)
+                        }
+                    }
+                }
+            }
+
+            if (pullRefreshState.isRefreshing) {
+                LaunchedEffect(key1 = true) {
+                    onRefresh()
+                }
+            }
+            LaunchedEffect(key1 = isRefreshing) {
+                if (isRefreshing)
+                    pullRefreshState.startRefresh()
+                else
+                    pullRefreshState.endRefresh()
+            }
+
+            PullToRefreshContainer(
```

Comment:
have you tried Indicator instead of Container

Reply:
The Indicator and Pasing isrefresh and refresh state insite pull refresh state is when we use modifier.pullrefresh() in material, but i am implementing it using nestedScrollConnection that is new pull refresh state on material 3 library